src/core/public/i18n/i18n_eui_mapping.tsx

```diff
@@ -272,9 +272,11 @@ export const getEuiContextMapping = (): EuiTokensObject => {
     'euiColumnSorting.button': i18n.translate('core.euiColumnSorting.button', {
       defaultMessage: 'Sort fields',
     }),
-    'euiColumnSorting.buttonActive': i18n.translate('core.euiColumnSorting.buttonActive', {
-      defaultMessage: 'fields sorted',
-    }),
+    'euiColumnSorting.buttonActive': ({ numberOfSortedFields }: EuiValues) =>
+      i18n.translate('core.euiColumnSorting.buttonActive', {
+        defaultMessage: '{numberOfSortedFields, plural, one {# field} other {# fields}} sorted',
```

Comment:
@afharo - does this pluralization look correct to you (following https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/GUIDELINE.md#pluralization)?

I've used this in the past in Enterprise Search, but this appears to be the first time in the EUI i18n translations that this format is used, so I wanted to double check!

Reply:
So unfortunately it looks like this is failing `i18n_eui_mapping.test.ts`:

<img width="1062" alt="" src="https://user-images.githubusercontent.com/549407/130701827-7558a17d-2016-44f0-b3a8-e45fbfe542a8.png">

Here's the exported `defString` of this token:

https://github.com/elastic/eui/blob/d0d7d501854cf24fd571e49c30f608e793268c20/i18ntokens.json#L1204

Unfortunately, even without the conversion of this token to Kibana's pluralization guidelines, I don't think this change would pass without _literally_ passing in a stringified function. Per @chandlerprall, because EUI is agnostic towards specific localization services/libraries, our `defaultMessage` would necessarily differ if there is anything more than straight variable interpolation.

@Bamieh, as a fix for this, is there any chance we can update `i18n_eui_mapping.test.ts` to either skip specified tokens, or to skip tokens that interpolate to functions? Here's my proposal with the latter implementation: https://github.com/elastic/kibana/pull/109926/commits/670b00d6bbb05a844932290d3e2081bcfdf8b4f5

Reply:
Sadly I dont see an ideal solution to this without adding the `values` or `variables` being used to the tokens json file so we can verify they are also used by the `i18n.translate` message.

I think it is OK to skip checking message values for functions until we work on a better approach to handling this.

We'd still be verifying that the message ids are defined for these labels so at least we still have the guarantee of full coverage.

I'd recommend using `/^\(.*?\).*?=>/` instead of checking for `=>` alone as it might skip unexpected messages.

Reply:
Thanks @Bamieh, super appreciate the flexibility 🙇‍♀️ I've made the suggested regex change here.

@afharo, that's correct. @chandlerprall can likely clarify/correct me if I'm wrong, but as far as I understand it, EUI itself is not responsible for and does not perform any i18n in and of itself. It merely exposes our text/strings as tokens that other apps can then hook into to pass their own i18n services/functions (so in this case in Kibana's token mapping file, I made sure that the pluralization matched the format that EUI generally uses).

So any "i18n" that EUI does is for its English default language (which must at least work/be correct for apps that choose not to localize). In which case the `s` is OK for the default English, but other apps ingesting EUI that want to have (e.g.) Italian translations must then override the token's defaultMessage with their own translation and pluralization that fits the library they're using. Hope that makes sense!

> Skipping validation of functions in the tests seems to me like a hack to avoid being consistent.

I agree that generally skipping isn't ideal long-term and I'd love to see a follow-up PR after this one. However, I disagree slightly in that I think the premise of the test itself is flawed and potentially a little too rigid. Because EUI is not in and of itself responsible for i18n, I personally don't think the test should be checking for 1:1 `defString` matching - as we mentioned, our defaultMessage _is_ going to necessarily differ from production apps with actual i18n services on anything with complex sentence structures/pluralization/anything more than straight variable interpolation.

I'm not too sure what the best answer would be, but my suggestion would might be to either only check for the existence of a mapping (i.e. ensuring a token wasn't forgetten) and not for translation equality, or again just skipping functions because those are likely going to be the ones with more complex structure and will need custom i18n handling.

Reply:
@constancecchen I can understand EUI's goal is not to officially support any other language, and it just provides the i18n capabilities for any downstream users to overwrite it.

However, as @chandlerprall and you claim, it's a design system that ensures that it works correctly in English. Hence our tests are checking that the `defString` matches. If EUI changes the font of a component because its suggestion in English is to use different wording, our tests should catch that change and let us know. 

An example of that is the PR where these tests were originally introduced (#106377): we noticed `New-Old` and `Old-New` texts were flipped.

If we know that a specific token is intentionally different to EUI for any reason, I'd rather have it explicitly skipped in the tests (and commenting why), instead of skipping all the function-based checks. Especially because most of them are like `euiNotificationEventMeta.contextMenuButton` where the parameter is just another string to add in the token (`defString: "Menu for {eventName}"`).

What do you think?